WSL/disk-space.md

```diff
@@ -0,0 +1,217 @@
+      mount | grep ext4
+   ```
+
+   - You may see this message in response to the first **mount** command: "/dev: none already mounted on /dev." This message can safely be ignored.
+   - Copy the name of this entry, which will look like: `/dev/sdX` (with the X representing any other character).  In the following example the value of **X** is **b**:
+
+   ```bash
+      sudo resize2fs /dev/sdb <sizeInMegabytes>M
+   ```
+
+   - Using the example from above, we changed the vhd size to **512000**, so the command would be: `sudo resize2fs /dev/sdb 512000M`.
+
+   > [!NOTE]
+   > You may need to install **resize2fs**.  If so, you can use this command to install it:  `sudo apt install resize2fs`.
+
+   The output will look similar to the following:
+
+   ```bash
+      resize2fs 1.44.1 (24-Mar-2021)
+      Filesystem at /dev/sdb is mounted on /; on-line resizing required
+      old_desc_blocks = 32, new_desc_blocks = 38
+      The filesystem on /dev/sdb is now 78643200 (4k) blocks long.
+      ```
+
+> [!IMPORTANT]
+> We recommend that you do not modify, move, or access the WSL related files located inside of your AppData folder using Windows tools or editors. Doing so could cause your Linux distribution to become corrupted. If you would like to access your Linux files from Windows, that is possible via the path `\\wsl$\<distroName>\`. Open your WSL distribution and enter `explorer.exe .` to view that folder. To learn more, see the blog post: [Accessing Linux files from Windows](https://devblogs.microsoft.com/commandline/whats-new-for-wsl-in-windows-10-version-1903/#accessing-linux-files-from-windows).
+
+## How to repair a VHD mounting error
+
+If you encounter an error related to "mounting the distribution disk", this could be due to a sudden shutdown or power outage and may result in the Linux distribution VHD being switched to *read-only* to avoid data loss. You can repair and restore the distribution using the `e2fsck` Linux command by following the steps below.
+
+### Use the lsblk command to identify the block device name
+
+When WSL 2 installs a Linux distribution, it is mounting the distribution as a Virtual Hard Disk (VHD) with it's own file system. Linux refers to these hard drives as "block devices" and you can view information about them by using the `lsblk` command.
+
+To find the names of the block devices currently being used by WSL 2, open your distribution and enter the command: `lsblk`. (Or open PowerShell and enter the command: `wsl.exe lsblk`.) The output will look something like this:
+
+```bash
+NAME MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
+sda    8:0    0 363.1M  1 disk
+sdb    8:16   0     8G  0 disk [SWAP]
+sdc    8:32   0   1.5T  0 disk
+sdd    8:48   0     1T  0 disk /mnt/wslg/distro
+```
+
+Information about the block device includes:
+
+- **NAME**: The name assigned to the device will be sd[a-z], referring to the SCSI Disk with a letter designation for each disk being used. `sda` is always the system distribution.
```

Comment:
This section feels a bit like too much information for the purpose of this page (and might create extra confusion, especially if we talk about 'removable' or 'USB' devices here. I'd recommend just keeping '**NAME**' here

Reply:
Hmm. I found knowing these descriptions quite helpful and am frustrated when a command is run resulting in a list of information that I don't understand. I generally prefer to err on the side of explaining everything mentioned, but also am interested to better understand the potential for confusion... can you help clarify what you mean?

Reply:
This could be a good documentation to point to for `lslblk`: https://linuxhint.com/linux-lsblk-command-tutorial-for-beginners/ 

Overall, I think we should avoid documenting what standard Linux command do and focus on WSL-specific content, since  well known Linux commands are usually well documented online.

I'm worried that users will get confused if we start talking about USB devices or LVM since this could create confusion which scenarios like [this one](https://github.com/microsoft/WSL/issues/7566) which are orthogonal to this

Reply:
I removed reference to USB and LVM. We generally try to avoid linking out to 3rd party sites unless we're pretty confident of their ownship and longevity (sites like Ubuntu.com)... and in this case doesn't seem necessary.